board/freescale/mx6ullevk/mx6ullevk_siklu_hw.c

```diff
@@ -107,32 +111,39 @@ SKL_BOARD_TYPE_E siklu_get_board_type(void)
 		reg_val = readl((uint32_t*)reg_addr);
 		//printf("reg_val - 0x%x\n",reg_val);
 
-		val = ((reg_val & 1<<18)?(1):(0)) + ((reg_val & 1<<19)?(2):(0)) + ((reg_val & 1<<20)?(4):(0)) + ((reg_val & 1<<21)?(8):(0));
+		val = ((reg_val & 1<<18)?(1):(0)) + ((reg_val & 1<<19)?(2):(0))
+			+ ((reg_val & 1<<20)?(4):(0))
+			+ ((reg_val & 1<<21)?(8):(0));
+
```

Comment:
Now I see that this actually duplicates `get_siklu_board_id()` (renamed to `spl_get_siklu_board_id()` in this commit), and all `GPIOx_IOxx` macros. Can we have just one shared routine to detect the board?

Reply:
This copy of the routine does things like change the U-Boot environment that we don't do in the SPL. I don't think that there is much added value in extracting common parts of the code. See the comment at the top of this routine regarding the NFS boot.

Reply:
It would be nice to have a single routine that just detects the board type, and encapsulates the information of GPIOs listing and order. All other routines can use the shared code without bothering with the GPIO level.

Reply:
I tried to do this and ran into runtime problems that I do not have time to debug. It is not work the effort at this time.